dags/analyzer_helper/discourse/extract_raw_info.py

```diff
@@ -0,0 +1,230 @@
+        self.driver = self.neo4jConnection.connect_neo4j()
+        self.forum_endpoint = forum_endpoint
+        self.converter = DateTimeFormatConverter()
+
+    def close(self):
+        """
+        Close the Neo4j connection.
+        """
+        self.driver.close()
+
+    def fetch_post_details(self, created_at: Optional[str] = None, comparison: Optional[str] = None) -> list:
+        """
+        Fetch details of posts from the Discourse forum.
+
+        :param created_at: Optional datetime string to filter posts created after this date.
+        :param comparison: Optional comparison operator, either 'gt' for greater than or 'gte' for greater than or equal to.
+        :return: List of dictionaries containing post details.
+        """
+        where_clause = ""
+        if created_at and comparison:
+            operator = '>' if comparison == 'gt' else '>='
+            where_clause = f"WHERE post.createdAt {operator} $createdAt"
+
+        query = f"""
+        MATCH (forum:DiscourseForum {{endpoint: $forum_endpoint}})
+        WITH forum
+        MATCH (topic:DiscourseTopic {{forumUuid: forum.uuid}})
+        MATCH (topic)-[:HAS_POST]->(post:DiscoursePost)
+        {where_clause}
+        OPTIONAL MATCH (post)<-[:POSTED]-(author:DiscourseUser)
+        OPTIONAL MATCH (post)<-[:LIKED]-(likedUser:DiscourseUser)
+        OPTIONAL MATCH (post)-[:REPLY_TO]->(repliedPost:DiscoursePost)
+        RETURN
+          id(post) AS post_id,
+          id(author) AS author_id,
+          post.createdAt AS created_at,
+          author.name AS author_name,
+          collect(DISTINCT likedUser.id) AS reactions,
+          id(repliedPost) AS replied_post_id,
+          id(topic) AS topic_id
+        LIMIT 10
+        """
+
+        with self.driver.session() as session:
+            if created_at and comparison:
+                result = session.run(query, forum_endpoint=self.forum_endpoint, createdAt=created_at)
+            else:
+                result = session.run(query, forum_endpoint=self.forum_endpoint)
+            posts = result.data()
+
+        return posts
+
+    def fetch_post_categories(self, post_ids):
+        """
+        Fetch categories associated with given post IDs.
+
+        :param post_ids: List of post IDs.
+        :return: List of dictionaries containing post categories.
+        """
+        # We are matching topics by post.topicId
+        # query = """
+        # UNWIND $post_ids AS post_id
+        # MATCH (post:DiscoursePost)
+        # WHERE id(post) = post_id
+        # MATCH (topic:DiscourseTopic)
+        # WHERE id(topic) = post.topicId
+        # OPTIONAL MATCH (category:DiscourseCategory)-[:HAS_TOPIC]->(topic)
+        # RETURN
+        #   id(post) AS post_id,
+        #   id(category) AS category_id,
+        #   category.name AS category_name
+        # """
+        # We are matching topics by topic.HAS_POST post.id
+        query = """
+        UNWIND $post_ids AS post_id
```

Comment:
Doing an unwind operation on large amount of posts could be intensive. You could do something like this below

```
        MATCH (post:DiscoursePost)
        WHERE id(post) in $post_ids
        MATCH (topic:DiscourseTopic)-[:HAS_POST]->(post)
        OPTIONAL MATCH (category:DiscourseCategory)-[:HAS_TOPIC]->(topic)
        RETURN
        id(post) AS post_id,
        id(category) AS category_id,
        category.name AS category_name
```